### ISSUE TITLE
Enhancements of Consent Banner for login pages

### ISSUE THREAD

**Issue by HoustonDad** [rancher/dashboard#4719]
For clarity, here's what needs to be done: for 2.6.4

For the login consent banner, add an extra setting 'Require user to accept' - default is true.

For the consent banner shown on the login page (added in 2.6.3), if this flag is false, we display the banner as is, otherwise we display it differently - we overlay a glass panel with subtle transparency and on top of that a panel to show the consent banner with an 'Accept' button. When the accept button is pressed, we remove the consent banner and glass overlay so that the user can then interact with the content beneath (i.e. they can then login).

--

**Setup**
- Rancher version: 2.6
- Browser type & version: N/A

**Describe the issue**
A new feature was recently added in Rancher 2.6 that allows orgs to add consent banners to their login pages. This is a great first iteration! 

https://github.com/rancher/dashboard/issues/4336

Many Government agencies require some sort of pop-up presented before the target application becomes usable, that requires the user to 'Accept' or hit 'OK' before any part of the application becomes usable. This is a DoD requirement based upon their STIGs 

*To Reproduce* 

WHEN a user logs into Rancher THEN that user is presented with a consent banner and a login box Result The user logging in does not have to 'Accept' the consent Expected Result GIVEN That an org is on the newest version of Rancher that provides this Consent Banner functionality, GIVEN that the org has set a 'consent' banner.

*Result* 
The user logging in does not have to 'Accept' the consent 

**Expected Result**
WHEN a user logs into Rancher THEN that user is presented with a consent banner, and is forced to either accept or deny the consent 

Accepting the consent should display the login area. 
Denying the consent could return a generic 403 or return to the Rancher default homepage.

**Additional context**
SYSTEM USE NOTIFICATION N/A N/A CCI-000048 SRG-APP-000068-CTR-000120 The container platform must display the Standard Mandatory DoD Notice and Consent Banner before granting access to platform components. 

SESSION TERMINATION | USER-INITIATED LOGOUTS N/A N/A CCI-002364 SRG-APP-000297-CTR-000705 Access to the container platform must display an explicit logout message to user indicating the reliable termination of authenticated communication sessions.

**Comment by nwmac** [comment]
@HoustonDad Can you clarify if the consent must be acknowledged before or after login - in the expected result it says the user logs into Rancher and then they get the consent banner - if they accept it they then go to the login area. That does not make sense.

What makes sense is:

Users loads the Rancher UI in their browser and gets the login page. If configured, a consent panel is overlaid on the login page with an accept button which the user must press to dismiss the consent panel and allow them to enter the login details in order to login.

**Comment by HoustonDad** [comment]
@nwmac We've discussed this a bit in RFED this week. 

We've been able to find current implementations that have the consent banner before logging in. 

I think we will be able to meet requirements with your story
> Users loads the Rancher UI in their browser and gets the login page. If configured, a consent panel is overlaid on the login page with an accept button which the user must press to dismiss the consent panel and allow them to enter the login details in order to login.

Thanks for following up!

**Comment by gaktive** [comment]
The original case description has been clarified based on new requirements (and it was way too confusing to break down the changes into a single comment).

**Comment by nwmac** [comment]
# How to test

This is an enhancement that adds a new configuration option to show a consent dialog on the login screen rather than presenting the consent message as banner.

This is configured from Global Settings > Branding.

Go to Global Setting > Branding. Check 'Show Consent Banner on Login Screen'.

There is now a new Checkbox 'Show Login Consent as a modal dialog' and an input control 'Accept Button text'.

Check the checkbox 'Show Login Consent as a modal dialog' and enter some text in the accept button field. Enter a message in the 'text' input box. Click 'Apply'. Logout and you should see a dialog with your consent massage - you must click the button before you can login.

Also check that you can configure the consent banner without the 'Show login consent as a modal dialog' checked, and that in this mode, when you logout, the  message is displayed as a banner and not a dialog.

**Comment by jmcsagdc** [comment]
Verified in v2.6-da382a88600d0272d48537c5de88b3ce0ede6dd3-head

Test steps as above https://github.com/rancher/dashboard/issues/4719#issuecomment-1027243196

I went a little further by setting font attributes and colors too. Used modal and banner approaches. Both work as expected.

![2022-02-01_12-52-42](https://user-images.githubusercontent.com/15971368/152049597-d93ef786-cee1-49f7-bb5d-366f3a7ed5dd.png)
